source/common/http/conn_pool_base.cc

```diff
@@ -89,5 +341,65 @@ void ConnPoolImplBase::onPendingRequestCancel(PendingRequest& request) {
   checkForDrained();
 }
 
+namespace {
+// Translate zero to UINT64_MAX so that the zero/unlimited case doesn't
+// have to be handled specially.
+uint64_t translateZeroToUnlimited(uint64_t limit) {
+  return (limit != 0) ? limit : std::numeric_limits<uint64_t>::max();
+}
+} // namespace
+
+ConnPoolImplBase::ActiveClient::ActiveClient(ConnPoolImplBase& parent,
+                                             uint64_t lifetime_request_limit,
+                                             uint64_t concurrent_request_limit)
+    : parent_(parent), remaining_requests_(translateZeroToUnlimited(lifetime_request_limit)),
+      concurrent_request_limit_(translateZeroToUnlimited(concurrent_request_limit)),
+      connect_timer_(parent_.dispatcher_.createTimer([this]() -> void { onConnectTimeout(); })) {
+  Upstream::Host::CreateConnectionData data = parent_.host_->createConnection(
+      parent_.dispatcher_, parent_.socket_options_, parent_.transport_socket_options_);
+  real_host_description_ = data.host_description_;
+  codec_client_ = parent_.createCodecClient(data);
+  codec_client_->addConnectionCallbacks(*this);
+
+  conn_connect_ms_ = std::make_unique<Stats::HistogramCompletableTimespanImpl>(
+      parent_.host_->cluster().stats().upstream_cx_connect_ms_, parent_.dispatcher_.timeSource());
+  conn_length_ = std::make_unique<Stats::HistogramCompletableTimespanImpl>(
+      parent_.host_->cluster().stats().upstream_cx_length_ms_, parent_.dispatcher_.timeSource());
+  connect_timer_->enableTimer(parent_.host_->cluster().connectTimeout());
+
+  parent_.host_->stats().cx_total_.inc();
```

Comment:
Lazy check - are all these stats tested in existing H2 tests?

Reply:
Missing test coverage:
* upstream_cx_active_

I'll add a case for it